Marlin/i2cEncoder.h

```diff
@@ -0,0 +1,254 @@
+          uint8_t errArrayIndex = 0;
+        #endif
+        
+    public:
+        void init(AxisEnum axis, byte address);
+        void update();
+        void set_homed();
+//        double get_axis_error_mm(bool report);
+        long get_axis_error_steps(bool report);
+        double get_position_mm();
+        double mm_from_count(long count);
+        long get_position();
+        long get_raw_count();
+//        void set_led_param(byte, byte, byte);
+        void set_zeroed();
+        bool passes_test(bool report,bool &moduleDetected);
+        bool passes_test(bool report);
+        byte get_magnetic_strength();
+        bool test_axis();
+        void calibrate_steps_mm(int iterations);
+
+        void set_current_position(float newPositionMm) ;
+
+        int get_error_count();
+        void set_error_count(int newCount);
+        
+        void set_address(byte address);
+        byte get_address();
+
+        void set_active(bool);
+        bool get_active();
+
+        void set_inverted(bool);
+        bool get_inverted();
+
+        AxisEnum get_axis();
+        void set_axis(AxisEnum axis);
+
+        bool get_error_correct_enabled();
+        void set_error_correct_enabled(bool enabled);
+
+        byte get_error_correct_method();
+        void set_error_correct_method(byte method);
+
+        float get_error_correct_threshold();
+        void set_error_correct_threshold(float newThreshold);
+
+        int get_encoder_ticks_mm();
+
+        int get_encoder_ticks_unit();
+        void set_encoder_ticks_unit(int ticks);
+
+        byte get_encoder_type();
+        void set_encoder_type(byte type);
+
+        int get_stepper_ticks();
+        void set_stepper_ticks(int ticks);
+
+        float get_axis_offset();
+        void set_axis_offset(float newOffset);
+};
+
+class EncoderManager {
+    private:
+        
+
+    public:
+        EncoderManager();
+        void init();
+        void update();
+        void homed(AxisEnum axis);
+        void report_position(AxisEnum axis, bool units, bool noOffset);
+        void report_status(AxisEnum axis);
+//        void report_error(AxisEnum axis);
+        void test_axis(AxisEnum axis);
+        void test_axis();
+        void calibrate_steps_mm(AxisEnum axis, int iterations);
+        void calibrate_steps_mm(int iterations);
+//        void change_module_address(int oldAddress, int newAddress);
+//        void check_module_firmware(int address);
+        void report_error_count(AxisEnum axis);
+        void report_error_count();
+        void reset_error_count(AxisEnum axis);
+        void reset_error_count();
+        void enable_error_correction(AxisEnum axis, bool enabled);
+        void set_error_correct_threshold(AxisEnum axis, float newThreshold);
+        void get_error_correct_threshold(AxisEnum axis);
+
+        int get_encoder_index_from_axis(AxisEnum axis);
+
+        I2cEncoder encoderArray[NUM_AXIS];
+
+};
+
+static inline int8_t sgn(int val) {
```

Comment:
Apologies - I somehow thought these were a combined comparison functions when I looked at them the first time.  I see now that they're just returning the sign of the variable.  Sorry about that.

So you'd want something like this: 
`#define SIGN(a) ((a>0) - (a<0))`

Can just use that in place of the sgn() functions; no need to worry about types, etc.

If that produces a compiler warning anywhere (I don't think it should), you can also use a pair of ternaries, which should be fine (but maybe a little slower):
#define SIGN(a) (a<0 ? -1 : (a>0 ? 1 : 0))